bundle/regal/rules/idiomatic/equals_pattern_matching.rego

```diff
@@ -0,0 +1,82 @@
+# METADATA
+# description: Prefer pattern matching in function arguments
+package regal.rules.idiomatic["equals-pattern-matching"]
+
+import future.keywords.contains
+import future.keywords.if
+import future.keywords.in
+
+import data.regal.ast
+import data.regal.result
+
+# Current limitations:
+# Only works for single comparison either in head or in body
```

Comment:
```rego
f(x) := x if x == 1
```
Will this pass or fail?

Reply:
It will fail, as it could be rewritten, although that case is arguable contrived:
```
f(1) := 1
```
I added a unit test to verify just now:
https://github.com/StyraInc/regal/pull/411/files#diff-2800e3666d00c08742a4a86e8a7557117cd8ee69172cd5bcc6f3fff410f79a3cR50-R55

Do you think we should ignore that?

Reply:
This is fine, thanks!